src/Umbraco.Cms.Integrations.Commerce.Shopify/Client/src/property-editor/shopify-product-picker-property-editor.element.ts

```diff
@@ -123,7 +123,7 @@ export class ShopifyProductPickerPropertyEditor extends UmbLitElement implements
 
     deleteProduct(id: number) {
         var index = this.products.map(p => p.id).indexOf(id);
-        this.products.splice(index);
+        this.products = this.products.filter(product => product.id != id);
```

Comment:
This looks right to me.  If the collection is immutable you'll have to replace it with a new collection without the item you want to remove.